modules/Bio/EnsEMBL/DBSQL/DBAdaptor.pm

```diff
@@ -1051,6 +1024,88 @@ sub to_hash {
   return $hash;
 }
 
+#########################
+# Switchable adaptor methods
+#########################
+
+=head2 switch_adaptor
+
+  Arg [1]     : String name of the adaptor type to switch out
+  Arg [2]     : Reference The switchable adaptor implementation
+  Arg [3]     : (optional) CodeRef Callback which provides automatic switchable adaptor cleanup 
```

Comment:
Does this mean the callback has to do cleanup, or that cleanup is done after the callback is run?